### PR TITLE
Add throttling on requests.

### PR DESCRIPTION
**Pull request by dliusquare**

### CODE REVIEW

File: config/initializers/rack_attack.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+RACK_ATTACK_CONNECTION_POOL = ConnectionPool.new(size: 10, timeout: 0.1) do
+  redis = Redis.new(Shuttle::Configuration.redis.symbolize_keys)
+  Redis::Namespace.new('rack::attack', redis: redis)
+end
+
+Rack::Attack.cache.store = RateLimitRedisCache.new(RACK_ATTACK_CONNECTION_POOL)
+
+# Lockout IP addresses that are hammering user sign ups, sign ins and unlocks.
+Rack::Attack.blacklist('allow2ban lead form scrappers') do |req|
```
Comment (mrpoptart) [inline config/initializers/rack_attack.rb]: Should this be "scrapers"?

File: config/initializers/rack_attack.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    # This is used to experiment what the proper IP should be used in throttling.
+    ip = req.ip
+    client_ip = req.env['HTTP_CLIENT_IP']
+    forward_ip = req.env['HTTP_X_FORWARDED_FOR']
+    remote_addr = req.env['REMOTE_ADDR']
+    Rails.logger.info("RACK-ATTACK checking - ip:#{ip} client_ip:#{client_ip} forward_ip:#{forward_ip} remote_addr:#{remote_addr}")
+
+    # After 20 requests in 5 minute, block all requests from that IP for 1 hour.
+    real_ip = client_ip || remote_addr || forward_ip || ip || 'unknown ip'
+    Rack::Attack::Allow2Ban.filter(real_ip, maxretry: 20, findtime: 5.minutes, bantime: 1.hour) do
```
Comment (mrpoptart) [inline config/initializers/rack_attack.rb]: i think this is fine, but maybe we should limit signups more? Do we imagine that someone's going to have to sign up more than an couple times in an hour? The only situation I can think of is if we onboard a bunch of new translators and they all sign up from the same computer at once. Even then, it seems unlikely.

File: spec/controllers/registrations_controller_spec.rb [hunk 3]
```diff
@@ -14,35 +14,157 @@
...
+      let(:extra_params) { { 'REMOTE_ADDR' => '102.0.0.2' } }
+
+      it 'failed to login' do
+        response = subject
+        expect(response).to eq(200)
+      end
+    end
+
+    context 'throttling requests' do
+      let(:extra_params) { { 'REMOTE_ADDR' => '102.0.0.3' } }
```
Comment (mrpoptart) [inline spec/controllers/registrations_controller_spec.rb]: We should also check that someone with a different IP can still proceed even if someone else is throttled.

File: config/initializers/rack_attack.rb (hunk 2 above)
Comment (dliusquare) [reply config/initializers/rack_attack.rb]: Currently the 20 counts are combined for all operations of sign up, unlock and sign in.

**Pull request merged**